File: back-end-scala/src/main/scala/com/idak/tuto/api/resource/UserResource.scala
package com.idak.tuto.api.resource

import com.idak.tuto.api.model.User
import com.idak.tuto.api.service.UserService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation._

@RestController
@RequestMapping(path = Array("/users"))
class UserResource(userService: UserService) {

  @GetMapping
  def get = userService.get

  @GetMapping(path = Array("/{id}"))
  def get(@PathVariable id: Int) : User = userService.getOne(id).get

  @PostMapping
  def add(@RequestBody user: User) : User = userService.create(user)

  @DeleteMapping(Array("/{id}"))
  @ResponseStatus(HttpStatus.NO_CONTENT)
  def delete(@PathVariable id: Int) = userService.delete(id)

}

File: back-end-java/src/main/java/com/idak/tuto/api/resource/UserResource.java
package com.idak.tuto.api.resource;

import com.idak.tuto.api.model.User;
import com.idak.tuto.api.service.UserService;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/users")
public class UserResource {
    private UserService userService;

    public UserResource(UserService userService) {
        this.userService = userService;
    }

    @GetMapping
    public List<User> get(){
        return this.userService.get();
    }

    @GetMapping("/{id}")
    public User get(@PathVariable int id){
        return this.userService.get(id);
    }

    @PostMapping
    @ResponseStatus(HttpStatus.CREATED)
    public User save(@RequestBody User user){
        return this.userService.create(user);
    }

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    public void delete(@PathVariable int id){
        this.userService.delete(id);
    }

    @PutMapping
    @ResponseStatus(HttpStatus.ACCEPTED)
    public void update(@RequestBody User user){
        this.userService.update(user);
    }
}
